DbAccessor/src/Posts/FindPost.cs

```diff
@@ -0,0 +1,140 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+using MySql.Data.MySqlClient;
+
+namespace GoRideShare.posts
+{
+    // This class handles searching for a Post
+    public class FindPost(ILogger<FindPost> logger)
+    {
+        private readonly ILogger<FindPost> _logger = logger;
+
+        // This function is triggered by an HTTP GET request to create a new post
+        [Function("PostsSearch")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "Posts/Search")] HttpRequest req)
+        {
+            // Read the request body to get the 'Search Criteria'
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            SearchCriteria? searchCriteria;
+            try
+            {
+                searchCriteria = JsonSerializer.Deserialize<SearchCriteria>(requestBody);
+                if (searchCriteria != null) {
+                    var (invalid, errorMessage) = searchCriteria.validate();
+                    if (invalid)
+                    {
+                        _logger.LogError($"Search Criteria are not valid: {errorMessage}");
+                        return new BadRequestObjectResult(errorMessage);
+                    }
+                } else {
+                    _logger.LogError("Input was null");
+                    return new BadRequestObjectResult("Input was null");
+                }
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogError($"JSON deserialization failed: {ex.Message}");
+                return new BadRequestObjectResult("Incomplete Search Criteria.");
+            }
+            _logger.LogInformation($"Raw Request Body: {requestBody}");
+
+            // Retrieve the database connection string from environment variables
+            string? connectionString = Environment.GetEnvironmentVariable("DB_CONNECTION_STRING");
+            using (var connection = new MySqlConnection(connectionString))
+            {
+                var (error, message) = await Utilities.ValidateConnection(connectionString, connection);
+                if (error)
+                {
+                    _logger.LogError(message);
+                    return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+                }
+
+                // Use a parameterized query to fetch the posts
+                var query = """
```

Comment:
Here in this query, we're not joining the `users` table to the `posts`, which ended up giving error i.e. `user_id` not found and other columns from the `users` table not found. I will fix that and push the changes to the same PR.